core/src/main/java/org/infinispan/remoting/rpc/RpcManagerImpl.java

```diff
@@ -421,20 +426,20 @@ private <T> T rethrowAsCacheException(Throwable throwable) {
       }
    }
 
-   private CacheRpcCommand toCacheRpcCommand(ReplicableCommand command) {
-      checkTopologyId(command);
-      var cmd =command instanceof CacheRpcCommand ?
-            (CacheRpcCommand) command :
-            cf.wired().buildSingleRpcCommand((VisitableCommand) command);
-      setClusterTraceSpanAttributes(cmd);
-      return cmd;
+   private CacheRpcCommand initCacheRpcCommand(ReplicableCommand command) {
+      if (command instanceof CacheRpcCommand cmd) {
+         cmd.setCacheName(cacheName);
+         setClusterTraceSpanAttributes(cmd);
+         return cmd;
+      }
+      throw new IllegalStateException("Command must be an instanceof CacheRpcCommand: " + command);
```

Comment:
The `RpcManager` is a public API, but I would change the methods to `CacheRpcCommand` instead of `ReplicableCommand`. It is better to have a compile error than a runtime exception.